rules/manager.go

```diff
@@ -267,22 +268,22 @@ type Group struct {
 
 	metrics *Metrics
 
-	ruleGroupPostProcessFunc RuleGroupPostProcessFunc
+	// rule group evaluation iteration function, which by default invokes DefaultEvalIterationFunc
+	evalIterationFunc GroupEvalIterationFunc
 }
 
-// This function will be used before each rule group evaluation if not nil.
-// Use this function type if the rule group post processing is needed.
-type RuleGroupPostProcessFunc func(g *Group, lastEvalTimestamp time.Time, log log.Logger) error
+// Function type for extending rule group evaluation iteration logic
```

Comment:
It might be nice to explain a little more the expected API here, I know a user of this has to be fairly familiar with Prometheus, but for example `g.metrics` and the `.set*` functions won't be available outside this package. It seems a bit like `iter` has been factored out as it happened to be in the code, but that doesn't exactly make sense for an external hook.

Reply:
I've expanded on the comments in the latest revision:

https://github.com/prometheus/prometheus/blob/cbaf45edf3e683b507d67ddba71ddd51ea564292/rules/manager.go#L275-L280